File: todolist/src/types/task.ts
// ====== 任务状态 ======
export type TaskStatus = 'pending' | 'in_progress' | 'review' | 'completed' | 'cancelled' | 'paused';

// ====== 优先级 ======
export type TaskPriority = 0 | 1 | 2 | 3; // P0=紧急 P1=重要 P2=普通 P3=低优

export const PRIORITY_LABELS: Record<number, string> = {
  0: 'P0 紧急',
  1: 'P1 重要',
  2: 'P2 普通',
  3: 'P3 低优',
};

export const PRIORITY_COLORS: Record<number, string> = {
  0: '#E03131',
  1: '#F76707',
  2: '#4C6EF5',
  3: '#868E96',
};

// ====== 任务实体 ======
export interface Task {
  id: string;
  title: string;
  description: string;
  status: TaskStatus;
  priority: TaskPriority;
  estimated_minutes: number | null;
  actual_minutes: number | null;
  due_date: string | null;
  scheduled_date: string | null;
  is_recurring: boolean;
  recurring_rule: string | null;
  parent_task_id: string | null;
  sort_order: number;
  is_mit: boolean;
  completion_note: string;
  created_at: string;
  updated_at: string;
  completed_at: string | null;
  start_date: string | null;
  end_date: string | null;
  progress: number;
  color?: string;
}

// ====== 分类 ======
export interface Category {
  id: string;
  name: string;
  color: string;
  icon: string;
  sort_order: number;
  created_at: string;
}

// ====== 标签（多级） ======
export interface Tag {
  id: string;
  name: string;
  parent_tag_id: string | null;
  color: string;
  created_at: string;
  children?: Tag[];
}

// ====== 创建/更新参数 ======
export interface CreateTaskParams {
  title: string;
  description?: string;
  priority?: number;
  estimated_minutes?: number | null;
  due_date?: string | null;
  scheduled_date?: string | null;
  parent_task_id?: string | null;
  category_id?: string | null;
  start_date?: string | null;
  end_date?: string | null;
  progress?: number;
  color?: string;
}

export interface UpdateTaskParams {
  title?: string;
  description?: string;
  status?: TaskStatus;
  priority?: number;
  estimated_minutes?: number | null;
  actual_minutes?: number | null;
  due_date?: string | null;
  scheduled_date?: string | null;
  is_recurring?: boolean;
  recurring_rule?: string | null;
  category_id?: string | null;
  completion_note?: string | null;
  start_date?: string | null;
  end_date?: string | null;
  progress?: number;
  color?: string;
}

export interface TaskFilter {
  status?: TaskStatus;
  category_id?: string;
  search?: string;
  scheduled_date?: string;
  page?: number;
  page_size?: number;
}

export interface TaskPage {
  tasks: Task[];
  total: number;
  page: number;
  page_size: number;
}

// ====== 任务依赖关系 ======
export interface TaskRelationship {
  id: string;
  source_task_id: string;
  target_task_id: string;
  relationship_type: 'depends_on' | 'related_to';
  is_blocking: boolean;
  label: string;
  created_at: string;
}

export interface CreateTaskRelationshipParams {
  source_task_id: string;
  target_task_id: string;
  relationship_type?: string;
  is_blocking?: boolean;
  label?: string;
}

File: todolist/src-tauri/src/models/task.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Task {
    pub id: String,
    pub title: String,
    pub description: String,
    pub status: String,
    pub priority: i32,
    pub estimated_minutes: Option<i32>,
    pub actual_minutes: Option<i32>,
    pub due_date: Option<String>,
    pub scheduled_date: Option<String>,
    pub is_recurring: bool,
    pub recurring_rule: Option<String>,
    pub parent_task_id: Option<String>,
    pub sort_order: i32,
    pub is_mit: bool,
    pub completion_note: String,
    pub created_at: String,
    pub updated_at: String,
    pub completed_at: Option<String>,
    pub start_date: Option<String>,
    pub end_date: Option<String>,
    pub progress: i32,
    pub color: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct CreateTaskParams {
    pub title: String,
    #[serde(default)]
    pub description: String,
    #[serde(default = "default_priority")]
    pub priority: i32,
    pub estimated_minutes: Option<i32>,
    pub due_date: Option<String>,
    pub scheduled_date: Option<String>,
    pub parent_task_id: Option<String>,
    pub category_id: Option<String>,
    pub start_date: Option<String>,
    pub end_date: Option<String>,
    #[serde(default)]
    pub progress: i32,
    #[serde(default)]
    pub color: Option<String>,
}

fn default_priority() -> i32 {
    2
}

#[derive(Debug, Deserialize)]
pub struct UpdateTaskParams {
    pub title: Option<String>,
    pub description: Option<String>,
    pub status: Option<String>,
    pub priority: Option<i32>,
    pub estimated_minutes: Option<i32>,
    pub actual_minutes: Option<i32>,
    pub due_date: Option<String>,
    pub scheduled_date: Option<String>,
    pub is_recurring: Option<bool>,
    pub recurring_rule: Option<String>,
    pub parent_task_id: Option<String>,
    pub sort_order: Option<i32>,
    pub is_mit: Option<bool>,
    pub category_id: Option<String>,
    pub completion_note: Option<String>,
    pub start_date: Option<String>,
    pub end_date: Option<String>,
    pub progress: Option<i32>,
    pub color: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct TaskFilter {
    pub status: Option<String>,
    pub category_id: Option<String>,
    pub search: Option<String>,
    pub scheduled_date: Option<String>,
    #[serde(default = "default_page")]
    pub page: i32,
    #[serde(default = "default_page_size")]
    pub page_size: i32,
}

fn default_page() -> i32 {
    1
}
fn default_page_size() -> i32 {
    50
}

#[derive(Debug, Serialize)]
pub struct TaskPage {
    pub tasks: Vec<Task>,
    pub total: i32,
    pub page: i32,
    pub page_size: i32,
}
